crowbar_framework/app/models/service_object.rb

```diff
@@ -257,20 +257,31 @@ def remove_pending_elements(bc, inst, elements)
     end
   end
 
+  def restore_node(node)
+    node.crowbar['state'] = 'ready'
+    node.crowbar['state_owner'] = ""
+    node.save
+  end
+
   def restore_to_ready(nodes)
     with_lock "BA-LOCK" do
       nodes.each do |node_name|
         node = NodeObject.find_node_by_name(node_name)
-        next if node.nil?
-
-        # Nothing to delay so mark them applying.
-        node.crowbar['state'] = 'ready'
-        node.crowbar['state_owner'] = ""
-        node.save
+        next if node.nil? or node.crowbar['state'] == 'upgrade'
```

Comment:
Oh, it's because you set the state here: https://github.com/crowbar/barclamp-updater/pull/11/files#diff-cefc874d9c651cd9f09e3e298163248eR62

That's probably wrong. Will think about it.